ui/src/reportWebVitals.ts

```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals';
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+      getCLS(onPerfEntry);  
+      getFID(onPerfEntry);
+      getFCP(onPerfEntry);
+      getLCP(onPerfEntry);
+      getTTFB(onPerfEntry);
+    });
+  }
+}
```

Comment:
@olevitt  I don't know if you wrote this code or if it's just some boilerplate that you copied but allow me to give you some feedback that you might find useful.

1) ReportHandler is a type alias it can be imported as such using `import type { ReportHandler } from "web-vitals"`. Unlike regular imports the type imports disappear when the code is transpired to JS.  
On top of that, if you don't use type import what is the point of using lazy import later on? The lib has already been imported anyway.
2) Try do deal with the edge case first, the real meat of the function should not be nested:
Don't
```typescript
function myFunc(){
  if( !edgeCase1 ){
     //meat of the function
 }
}
```
Do: 
````typescript
function myFunc(){
  if( edgeCase1 ){
     return; //Or throw or whatever
   }
  // The meat
}
````
3) `onPerfEntry` is of type `undefined | ReportHandler` it can't be anything else, you do not need to check if is is a function. You should write `if( onPerfEntry )` or `if( onPerfEntry !== undefined )`. You might think that in typescript unlike in Java types are only annotations and do not garenty  anything. It is true but the runtimes checks like this should only be performed at the edge of the system when the inputs are not trusted.  

4) This function does nothing if `onPreferEntry` is not passed. This arguement should be mandatory.